app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,21 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, general info, callstack
+    def self.deprecation_warning(foreman_version_deadline, options = {})
+      raise Foreman::Exception, _("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A\d[.]\d+[.]\d\z/)
+      controller = options[:controller]
+      method = options[:method]
+      info = options[:info]
+      callstack = options[:callstack] || caller
+      msg = _("You are using deprecated behavior, it will be removed in version %{version_deadline}.\n") % {:version_deadline => foreman_version_deadline}
+      msg += _("Use %{controller}#%{method} instead.\n") % {:controller => controller, :method => method} if controller && method
+      msg += "#{info}." if info
+      ActiveSupport::Deprecation.warn(_(msg), callstack)
```

Comment:
No need to re-translate an already translated string.

Usually log messages aren't translated, it would all be in English, so I'd suggest perhaps separating the log message and the exception - keep the detail in the English log message, but have a translated, shorter exception thrown without all of that info.